### PR TITLE
Remove extra guard for non-payable functions that use msg.value

### PR DESCRIPTION
**Pull request by leonardoalt**
Fixes the solc-js tests for https://github.com/ethereum/solidity/pull/4590

### CODE REVIEW

**Comment by leonardoalt** [comment]
@axic

File: test/DAO/DAO.sol [hunk 1]
```diff
@@ -409,14 +409,13 @@ contract DAO is DAOInterface, Token, TokenCreation {
...
         bool _newCurator
-    ) onlyTokenholders public returns (uint _proposalID) {
+    ) onlyTokenholders public payable returns (uint _proposalID) {
 
         // Sanity check
         if (_newCurator && (
             _amount != 0
             || _transactionData.length != 0
             || _recipient == curator
-            || msg.value > 0
```
Comment (axic) [inline test/DAO/DAO.sol]: Why remove this here?

Comment (leonardoalt) [reply test/DAO/DAO.sol]: Mistake

**Comment by leonardoalt** [comment]
Fixed and squashed

**Pull request merged**